datadog_checks_dev/datadog_checks/dev/tooling/commands/metadata.py

```diff
@@ -0,0 +1,138 @@
+
+    If `check` is specified, only the check will be validated, otherwise all matching files in directory.
+    """
+    check_list = []
+
+    root = get_root()
+
+    if check:
+        path = resolve_path(os.path.join(root, check))
+        if not dir_exists(path):
+            abort(
+                'Directory `{}` does not exist. Be sure to `ddev config set {repo} '
+                'path/to/integrations-{repo}`.'.format(path, repo=ctx.obj['repo_choice'])
+            )
+        else:
+            check_list = [check]
+    else:
+        check_list = [x for x in sorted(get_valid_checks())]
+
+    for current_check in check_list:
+        if current_check.startswith('datadog_checks_'):
+            continue
+
+        # get any manifest info needed for validation
+        manifest = load_manifest(current_check)
+        try:
+            metric_prefix = manifest['metric_prefix']
+        except KeyError:
+            echo_info('{}: metric_prefix does not exist in manifest'.format(current_check))
+            metric_prefix = None
+
+        path = resolve_path(os.path.join(root, current_check, METADATA_FILE))
+        if not file_exists(path):
+            echo_failure('Missing metadata file at: {}'.format(path))
+        else:
+            # To make logging less verbose, common errors are counted for current check
+            metric_prefix_count = dict()
+            empty_count = dict()
+            duplicate_set = set()
+
+            with open(path) as f:
+                reader = csv.DictReader(f, delimiter=',')
+
+                for row in reader:
+                    # duplicates
+                    if str(row) not in duplicate_set:
+                        duplicate_set.add(str(row))
+                    else:
+                        raw_row = ','.join([y for x, y in row.items()])
+                        echo_warning("{}: `{}` is a duplicate row".format(current_check, raw_row))
+
+                    # all headers exist, no invalid headers
+                    if set(row.keys()) != ALL_HEADERS:
+                        invalid_headers = set(row.keys()).difference(ALL_HEADERS)
+                        if invalid_headers:
+                            echo_failure('{}: Invalid column {}'.format(current_check, invalid_headers))
+                        missing_headers = ALL_HEADERS.difference(set(row.keys()))
+                        if missing_headers:
+                            echo_failure('{}: Missing columns {}'.format(current_check, missing_headers))
+                        continue
+
+                    # metric_name header
+                    if metric_prefix and not row['metric_name'].startswith(metric_prefix):
+                        prefix = row['metric_name'].rsplit(".")[0]
+                        metric_prefix_count[prefix] = metric_prefix_count.get(prefix, 0) + 1
+
+                    # metric_type header
+                    valid = {'', 'count', 'counter', 'distribution', 'gauge', 'rate'}
```

Comment:
Could you create a constant variable for this at the top?

Reply:
Also @ofek should we allow deprecated values? Where can we find the full list of values.
Finally, if a value is deprecated, then i think we should have some warning for it and suggesting a fix, what do you folks think?